pkg/controllers/namespacesync/references.go

```diff
@@ -43,7 +43,7 @@ func getReference(
 		return nil, errors.New("reference is not set")
 	}
 
-	obj, err := external.Get(ctx, cli, ref, ref.Namespace)
+	obj, err := external.Get(ctx, cli, ref)
```

Comment:
This changed upstream https://github.com/kubernetes-sigs/cluster-api/commit/a21faa013812da0ec0b00dc0d80600e61a670d50